### PR TITLE
 AddressBook: remove master tag list

### PR DESCRIPTION
**Pull request by yamidark**
```
We keep track of a master tag list which holds all the tags in
AddressBook.

As the master tag list was deemed unnecessary and may be too
complex for new developers to grasp, as discussed in
se-edu/addressbook-level4#753 and se-edu/addressbook-level4#794,
the master tag list has already been removed in AB4
by se-edu/addressbook-level4#825.

UniqueTagList was also deemed unnecessary and removed in
addressbook-level4 after the master tag list is removed.

As such, the master tag list and UniqueTagList class should also be
removed in the lower level projects.

Let's remove the master tag list in AddressBook and AdaptedAddressBook,
as well as the UniqueTagList class.
```

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-import java.util.*;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
 
 /**
  * Represents the entire address book. Contains the data of the address book.
  *
```
Comment (Zhiyuan-Amos) [inline src/seedu/addressbook/data/AddressBook.java]: This * can be deleted.

File: src/seedu/addressbook/storage/jaxb/AdaptedAddressBook.java [hunk 2]
```diff
@@ -55,11 +47,6 @@ public AdaptedAddressBook(AddressBook source) {
      * so we check for that.
      */
     public boolean isAnyRequiredFieldMissing() {
-        for (AdaptedTag tag : tags) {
-            if (tag.isAnyRequiredFieldMissing()) {
-                return true;
-            }
-        }
         for (AdaptedPerson person : persons) {
```
Comment (Zhiyuan-Amos) [inline src/seedu/addressbook/storage/jaxb/AdaptedAddressBook.java]: I think we can concise this method by using stream:
`return persons.stream().anyMatch(AdaptedPerson::isAnyRequiredFieldMissing);`

**Review by Zhiyuan-Amos — COMMENTED** [review]
[1/3]

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 3]
```diff
@@ -15,10 +16,10 @@
     Address getAddress();
 
     /**
-     * The returned TagList is a deep copy of the internal TagList,
+     * The returned TagSet is a deep copy of the internal TagSet,
```
Comment (Zhiyuan-Amos) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java]: [2/2] oops sorry. `TagSet` looks weird. {@code Set<Tag>} looks better :P

**Review by Zhiyuan-Amos — COMMENTED** [review]
[3/3]

image

From AB-Level4, I'm thinking we can put an asterisk at the `Tag` to show that a `Person` can have multiple tags.

**Review by Zhiyuan-Amos — CHANGES_REQUESTED** [review]
Changes requested.

File: src/seedu/addressbook/storage/jaxb/AdaptedAddressBook.java [hunk 4]
```diff
@@ -36,13 +31,7 @@ public AdaptedAddressBook() {}
...
     public AdaptedAddressBook(AddressBook source) {
         persons = new ArrayList<>();
-        tags = new ArrayList<>();
-        for (ReadOnlyPerson person : source.getAllPersons()) {
-            persons.add(new AdaptedPerson(person));
-        }
-        for (Tag tag : source.getAllTags()) {
-            tags.add(new AdaptedTag(tag));
-        }
+        source.getAllPersons().forEach(person -> persons.add(new AdaptedPerson(person)));
```
Comment (Zhiyuan-Amos) [inline src/seedu/addressbook/storage/jaxb/AdaptedAddressBook.java:34]: I was thinking of:
`source.getAllPersons().immutableListView().stream().map(AdaptedPerson::new).collect(Collectors.toList());`

But this requires `source.getAllPersons().immutableListView()` which violates some rule (can't quite remember the rule). Basically, you should only be doing `foo.getBaz()` and not `foo.getBaz().getBar()`.

Writing this down for documentation purposes :)

**Review by Zhiyuan-Amos — APPROVED** [review]
👍

**Review by damithc — APPROVED** [review]

**Pull request merged**